This patch is based on the commit described below, taken from upstream pull
request 3175, it was adapted for use in the Debian package by Peter Michael
Green.

commit 4446ec06aa33813c732e2e473742cd5f410e1906
Author: Ola x Nilsson <olani@axis.com>
Date:   Mon Mar 20 14:32:07 2023 +0100

    gnu: Update F_SETLK and F_SETLKW for 64-bit time
    
    Also add the F_[SG]ET*64 constants.

Index: libc/src/unix/linux_like/linux/gnu/b32/mod.rs
===================================================================
--- libc.orig/src/unix/linux_like/linux/gnu/b32/mod.rs
+++ libc/src/unix/linux_like/linux/gnu/b32/mod.rs
@@ -254,9 +254,6 @@ cfg_if! {
 
         pub const PTRACE_DETACH: ::c_uint = 11;
 
-        pub const F_SETLK: ::c_int = 8;
-        pub const F_SETLKW: ::c_int = 9;
-
         pub const F_RDLCK: ::c_int = 1;
         pub const F_WRLCK: ::c_int = 2;
         pub const F_UNLCK: ::c_int = 3;
@@ -300,9 +297,6 @@ cfg_if! {
 
         pub const PTRACE_DETACH: ::c_uint = 17;
 
-        pub const F_SETLK: ::c_int = 6;
-        pub const F_SETLKW: ::c_int = 7;
-
         pub const F_RDLCK: ::c_int = 0;
         pub const F_WRLCK: ::c_int = 1;
         pub const F_UNLCK: ::c_int = 2;
@@ -338,6 +332,21 @@ cfg_if! {
         pub const EFD_CLOEXEC: ::c_int = 0x80000;
     }
 }
+cfg_if! {
+    if #[cfg(target_arch = "sparc")] {
+        pub const F_SETLK: ::c_int = 8;
+        pub const F_SETLKW: ::c_int = 9;
+    } else if #[cfg(all(gnu_time64_abi, target_arch = "mips"))] {
+        pub const F_SETLK: ::c_int = 34;
+        pub const F_SETLKW: ::c_int = 35;
+    } else if #[cfg(gnu_time64_abi)] {
+        pub const F_SETLK: ::c_int = 13;
+        pub const F_SETLKW: ::c_int = 14;
+    } else {
+        pub const F_SETLK: ::c_int = 6;
+        pub const F_SETLKW: ::c_int = 7;
+    }
+}
 
 align_const! {
     #[cfg(target_endian = "little")]
